MarvelApiClient/src/test/java/com/karumi/MarvelApiClient/CharacterApiClientTest.java

```diff
@@ -0,0 +1,251 @@
+
+    characterApiClient.getAll(ANY_OFFSET, ANY_LIMIT);
+
+    assertRequestSentToContains("offset=" + ANY_OFFSET, "limit=" + ANY_LIMIT);
+  }
+
+  @Test(expected = IllegalArgumentException.class) public void shouldThrowExceptionWhenLimitIsZero()
+      throws Exception {
+    CharacterApiClient characterApiClient = givenCharacterApiClient();
+    enqueueMockResponse(200);
+
+    characterApiClient.getAll(ANY_OFFSET, INVALID_LIMIT);
+  }
+
+  @Test public void shouldReturnValidUrlWhenICallToGetAllWithBuilder() throws Exception {
+    CharacterApiClient characterApiClient = givenCharacterApiClient();
+    enqueueMockResponse(200);
+
+    CharactersQuery query = CharactersQuery.Builder.create()
+        .withName(ANY_NAME)
+        .withNameStartWith(ANY_START_NAME)
+        .withLimit(ANY_LIMIT)
+        .withOffset(ANY_OFFSET)
+        .withModifiedSince(getAnyDate())
+        .withOrderBy(CharactersQuery.OrderBy.NAME, false)
+        .addComic(ANY_COMIC_1)
+        .addComic(ANY_COMIC_2)
+        .addEvents(getAnyEvents())
+        .addSerie(ANY_SERIE)
+        .addStory(ANY_STORY)
+        .build();
+    characterApiClient.getAll(query);
+
+    assertRequestSentToContains("offset=" + ANY_OFFSET, "limit=" + ANY_LIMIT, "name=" + ANY_NAME,
+        "nameStartsWith=" + ANY_START_NAME, "modifiedSince=" + ANY_MODIFIED_SINCE,
+        "orderBy=" + ORDER_NAME_DESCEDANT_VALUE, "comics=1,2", "events=1,2", "series=1",
+        "stories=1");
+  }
+
+  @Test(expected = MarvelApiException.class) public void shouldThrowExceptionWhenApiReturnAnError()
+      throws Exception {
+    CharacterApiClient characterApiClient = givenCharacterApiClient();
+    enqueueMockResponse(401,
+        "{\"code\":\"InvalidCredentials\",\"message\":\"That hash, timestamp and key "
+        + "combination is invalid.\"}");
+
+    try {
+      characterApiClient.getAll(ANY_OFFSET, ANY_LIMIT);
+    } catch (MarvelApiException e) {
+      assertEquals("InvalidCredentials", e.getMarvelCode());
+      assertEquals("That hash, timestamp and key combination is invalid.", e.getMessage());
+      throw e;
+    }
+  }
+
+  @Test public void shouldReturnAValidResponseWhenCallGetAll() throws Exception {
+    CharacterApiClient characterApiClient = givenCharacterApiClient();
+    enqueueMockResponse("getCharacters.json");
+
+    MarvelResponse<CharactersDto> characters = characterApiClient.getAll(0, ANY_LIMIT);
+
+    assertBasicMarvelResponse(characters);
+    CharactersDto charactersDto = characters.getResponse();
+    assertEquals(10, charactersDto.getCount());
+    assertEquals(10, charactersDto.getLimit());
+    assertEquals(0, charactersDto.getOffset());
+    assertEquals(1485, charactersDto.getTotal());
+
+    CharacterDto firstCharacter = charactersDto.getCharacters().get(0);
+    assertIs3dManCharacter(firstCharacter);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void shouldReturnAnIllegalExceptionWhenCallGetCharacterWithoutId() throws Exception {
```

Comment:
If I'm not wrong, ``INVALID_CHARACTER_ID`` is an empty string. Based on this I'd change this name test to ``shouldNotAcceptEmptyCharacterIds`` and the invalid character id with an empty string ``""`` or a null value.